resources/migrations/001_update_migrations.yaml

```diff
@@ -5652,7 +5652,61 @@ databaseChangeLog:
             onDelete: CASCADE
 
   - changeSet:
-      id: v50.2024-01-10T03:27:30
+      id: v50.2024-02-26T16:48:05
+      author: johnswanson
+      comment: Prohibit creating new data permission paths
+      changes:
+        - sql:
+            dbms: postgresql
+            sql: >-
+              ALTER TABLE permissions ADD CONSTRAINT no_data_permissions CHECK (object NOT LIKE '%/db/%') NOT VALID;
```

Comment:
"Fun" fact - every database we use has a different way of adding a constraint that _should not_ be checked for validity before it's created (in other words, you can create the constraint and it'll be enforced on inserts/updates but doesn't affect existing rows).

- postgres: `NOT VALID`
- mysql: `NOT ENFORCED`
- h2: `NOCHECK`
- mariadb: no way to do this natively, I used triggers as a workaround